### PR TITLE
Platform variable cf_role

### PR DESCRIPTION
**Pull request by benjwadams**
Adds check for `cf_role` attribute within platform variables for the IOOS 1.2 checker.

### CODE REVIEW

File: compliance_checker/base.py [hunk 1]
```diff
@@ -229,15 +229,15 @@ def assert_true(self, test, message):
...
-    if not hasattr(dataset, name):
+    if not hasattr(base_context, name):
         return 0
 
     ret_val = 1
-    if getattr(dataset, name) in allowed_vals:
+    if base_context.getncattr(name) in allowed_vals:
         ret_val += 1
 
     return ret_val
```
Comment (daltonkell) [inline compliance_checker/base.py]: Can we get some type info in the doc strings?

File: compliance_checker/ioos.py [hunk 2]
```diff
@@ -442,6 +445,15 @@ def __init__(self):
...
         ]
 
+    def setup(self, ds):
+        self.platform_vars = self._find_platform_vars(ds)
+
+    def _find_platform_vars(self, ds):
+        plat_vars = ds.get_variables_by_attributes(platform=
+                                                   lambda p: isinstance(p, str))
+        return {ds.variables[var.platform] for var in plat_vars if
+                var.platform in ds.variables}
```
Comment (daltonkell) [inline compliance_checker/ioos.py]: Could `self._platform_vars` be reused here as well?
https://github.com/daltonkell/compliance-checker/blob/0cb51efade62b729af35828f7ba8d6b46d36a603/compliance_checker/ioos.py#L548-L554

File: compliance_checker/base.py [hunk 3]
```diff
@@ -268,22 +268,30 @@ def attr_check(kvp, ds, priority, ret_val, gname=None):
...
+        display_name = "attribute {} in variable {}".format(name, var_name)
+        base_context = ds.variables[var_name]
+    else:
+        display_name = name
+        base_context = ds
     if other is None:
         res = std_check(ds, name)
         if not res:
-            msgs = ["%s not present" % name]
+            msgs = ["%s not present" % display_name]
```
Comment (daltonkell) [inline compliance_checker/base.py]: It's probably worth converting these to format strings where they appear

**Review by daltonkell — CHANGES_REQUESTED** [review]

File: compliance_checker/ioos.py (hunk 2 above)
Comment (benjwadams) [reply compliance_checker/ioos.py]: Maybe, but this check does something different.  It gets the value from each platform attribute from variables and adds it to a set.  The check you wrote doesn't check whether the variable exists in the dataset.  It also adds the global platform attribute to the set, which it shouldn't, as it isn't used to reference variables.

**Review by daltonkell — APPROVED** [review]

**Pull request merged**